src/lib/codap-helper.ts

```diff
@@ -314,17 +332,45 @@ export class CodapHelper {
         });
       }
 
-      // After initial join we allow destructive syncing
-      if (!initialJoin) {
-        const staleAttributes = originalAttributes.filter(attrA => {
-          return !sharedAttributes.some(attrB => attrA.name === attrB.name);
-        });
+      // synchronize properties of existing attributes
+      originalAttributes.forEach(origAttr => {
+        const sharedAttr = sharedAttributes.find(attr => attr.name === origAttr.name);
+        if (sharedAttr) {
+          const origAttrProps = origAttr.attr as any;
+          const sharedAttrProps = sharedAttr.attr as any;
+          const propsToUpdate: any = {};
+          let changed = false;
+          // tslint:disable-next-line: forin
+          for (const prop in sharedAttrProps) {
```

Comment:
Is there any chance that the user's original attributes have props that are unset/non-existent in the incoming attribute, and we would want to change the original attribute to match?

Would it be simpler and more robust to do a `JSON.stringify` comparison, and, if there are any differences, simple set the whole attribute to the new props, and not worry about which specific props have changed?

Reply:
Good question -- I thought about that approach and it felt a bit heavy-handed, but maybe that's appropriate in this case.